base_layer/core/src/base_node/state_machine_service/states/listening.rs

```diff
@@ -295,6 +298,42 @@ fn determine_sync_mode(
             network_tip_accum_difficulty.to_formatted_string(&Locale::en),
         );
 
+        // If both the local and remote are pruned mode, we need to ensure that the remote pruning horizon is
+        // greater_equal to ours so that we can sync all the data from it. If the remote is a pruned mode, and
+        // we only require some data from it, we need to ensure that they can supply the data we need, as in their
+        // effective pruned horizon is greater than our local current chain tip.
+        let pruned_mode = local.pruning_horizon() > 0;
+        let pruning_horizon_check = network.claimed_chain_metadata().pruning_horizon() > 0 &&
+            network.claimed_chain_metadata().pruning_horizon() < local.pruning_horizon();
+        let pruning_height_check = network.claimed_chain_metadata().pruned_height() > local.height_of_longest_chain();
+        let sync_able_peer = match (pruned_mode, pruning_horizon_check, pruning_height_check) {
```

Comment:
This is a really complicated way of specifying this. Is there not a simpler way?

Reply:
I think thats the simplest way. You are welcome to suggest a better one?
We only need to not sync to that peer on specific mode options on a truth table between remote pruning options, local pruning options.